pkg/serviceaccount/tokens_controller.go

```diff
@@ -44,18 +44,16 @@ type TokensControllerOptions struct {
 	// SecretResync is the time.Duration at which to fully re-list secrets.
 	// If zero, re-list will be delayed as long as possible
 	SecretResync time.Duration
-}
 
-// DefaultTokenControllerOptions returns
-func DefaultTokenControllerOptions(tokenGenerator TokenGenerator) TokensControllerOptions {
-	return TokensControllerOptions{TokenGenerator: tokenGenerator}
+	RootCAProvider func() []byte
```

Comment:
I won't ask you to change it back, but IMO making it a func() []byte is the sort of thing that could wait until we're actually putting code in to make a change here propagate. Because now it looks like it'll propagate, but in actuality it doesn't do any of the things you'd want it to, like actually changing service account secrets when this changes.